ktor-client/ktor-client-curl/src/io/ktor/client/engine/curl/Curl.kt

```diff
@@ -0,0 +1,19 @@
+package io.ktor.client.engine.curl
+
+import io.ktor.client.engine.*
+import libcurl.curl_global_init
+import libcurl.curl_global_cleanup
+import libcurl.CURL_GLOBAL_ALL
+
+object Curl : HttpClientEngineFactory<HttpClientEngineConfig> {
+    val code = curl_global_init(CURL_GLOBAL_ALL)
+
+    override fun create(block: HttpClientEngineConfig.() -> Unit): HttpClientEngine =
+        CurlClientEngine(CurlClientEngineConfig().apply(block))
+
+    fun close() {
```

Comment:
Please consider using a `Closeable` class instead.
Could we have 2 separate instances of  `Curl` engine?